### ISSUE TITLE
Errors when infer  using  the weights trained by myself

### ISSUE THREAD

**Issue by haoxurt** [haotian-liu/LLaVA#192]
### Discussion

I follow your instruction to finetune LLaVA-13B  with 8x A100 (80G).  The loss is normal.The curve  is as follows.
<img width="532" alt="image" src="https://github.com/haotian-liu/LLaVA/assets/43776429/4abcda8d-625a-4bc4-b844-1fdc0533235c">
My checkpoints is as follows:
<img width="737" alt="image" src="https://github.com/haotian-liu/LLaVA/assets/43776429/dd1c50fa-d632-42be-b463-db7dddd5223e">
Then I  excute the python3 -m llava.model.consolidate, and I get the new checkpoints as follows:
<img width="849" alt="image" src="https://github.com/haotian-liu/LLaVA/assets/43776429/2748e7b1-2b20-4498-b4b1-9469c5dfc8f5">
Finally ,I excute `python3 -m llava.model.apply_delta \
    --base /path/to/llama-13b \
    --target /output/path/to/LLaVA-13B-self \
    --delta path/to/llava-13b-finetune_consolidate`
and I get LLaVA-13B-self. It is as follows:
<img width="746" alt="image" src="https://github.com/haotian-liu/LLaVA/assets/43776429/b6df58eb-69f6-4261-8e61-5fcbac5c02a4">
When I python -m llava.eval.run_llava, I get the errors
<img width="820" alt="image" src="https://github.com/haotian-liu/LLaVA/assets/43776429/89c9a456-a2bc-4250-84c3-e14df4988709">
If I  use your  LLaVA-13b-delta-v0 ,it can work well.

**Comment by haoxurt** [comment]
I  have found a difference in config.json.  Is this the reason?
<img width="920" alt="image" src="https://github.com/haotian-liu/LLaVA/assets/43776429/6b1fc33b-404c-49f8-a137-ded69e41ba38">

**Comment by haotian-liu** [comment]
Hi @haoxurt 

You do *NOT* need to do `apply_delta` for the checkpoints that you trained.  This is only needed when you use a released LLaMA-based checkpoint (as its license does not allow us to release the full checkpoint).  Please try use the trained checkpoint directly, thanks!

**Comment by haoxurt** [comment]
Got it! Thanks for your quick reply!  
When I test using some simple image and query, its answer is wrong obviously. Could  you give some simple explanation or possible optimization strategy?
<img width="1256" alt="image" src="https://github.com/haotian-liu/LLaVA/assets/43776429/42585769-6839-4625-b541-63b162c3691d">
